File: javascript/smartlist/smartlist-api-mongodb-project/src/modules/report/dto/response/time-series-summary.resp.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { TimeSeriesDataRespDto } from './time-series-data.resp.dto';

export class TimeSeriesSummaryRespDto {
  @ApiProperty()
  attempts: TimeSeriesDataRespDto[];

  @ApiProperty()
  submissions: TimeSeriesDataRespDto[];

  constructor(
    attempts: TimeSeriesDataRespDto[],
    submissions: TimeSeriesDataRespDto[],
  ) {
    this.attempts = attempts;
    this.submissions = submissions;
  }
}

File: java/smartlist/src/main/java/com/smartlist/module/report/dto/response/TimeSeriesSummaryRespDTO.java
package com.smartlist.module.report.dto.response;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotEmpty;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class TimeSeriesSummaryRespDTO {

    @Schema(description = "Attempt details", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    final List<TimeSeriesDataRespDTO> attempts;

    @Schema(description = "Date details", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    final List<TimeSeriesDataRespDTO> submissions;

    public TimeSeriesSummaryRespDTO(final List<TimeSeriesDataRespDTO> attempts, final List<TimeSeriesDataRespDTO> submissions) {
        this.attempts = attempts;
        this.submissions = submissions;
    }


}
